### PR TITLE
Add CHANGELOG and integration tests for extra directories filtering 

### PR DESCRIPTION
**Pull request by chanseokoh**
For #2564 and #3164.

### CODE REVIEW

File: jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/SingleProjectIntegrationTest.java [hunk 1]
```diff
@@ -97,32 +94,29 @@ private static void assertLayerSize(int expected, String imageReference)
...
+    assertThat(dockerInspect)
+        .contains(
             "            \"Volumes\": {\n"
                 + "                \"/var/log\": {},\n"
                 + "                \"/var/log2\": {}\n"
-                + "            },"));
-    MatcherAssert.assertThat(
-        dockerInspect,
-        CoreMatchers.containsString(
+                + "            },");
```
Comment (mpeddada1) [inline jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/SingleProjectIntegrationTest.java:102]: optional: could we use regex matching to only test the values that are relevant?

**Review by mpeddada1 — COMMENTED** [review]
Could we add a test for maven as well?

File: jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/SingleProjectIntegrationTest.java (hunk 1 above)
Comment (chanseokoh) [reply jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/SingleProjectIntegrationTest.java:102]: I guess that involves a non-trivial amount of work, so maybe another time.

**Comment by chanseokoh** [comment]
Added a Maven integration test too.

I also realized a minor issue with the filtering. Will fix it in a follow-up PR.

File: jib-maven-plugin/src/integration-test/java/com/google/cloud/tools/jib/maven/BuildDockerMojoIntegrationTest.java [hunk 2]
```diff
@@ -99,12 +101,28 @@ public void testExecute_simple()
...
 
+  @Test
+  public void testExecute_simple_extraDirectoriesFiltering()
+      throws DigestException, IOException, InterruptedException, VerificationException {
+    String targetImage = "simpleimage:maven" + System.nanoTime();
+    buildToDockerDaemon(simpleTestProject, targetImage, "pom-extra-dirs-filtering.xml");
+    String output =
+        new Command("docker", "run", "--rm", "--entrypoint=ls", targetImage, "-1R", "/extras")
+            .run();
+    // No "bar" or "*.txt" files. Specifically,
```
Comment (mpeddada1) [inline jib-maven-plugin/src/integration-test/java/com/google/cloud/tools/jib/maven/BuildDockerMojoIntegrationTest.java:118]: how about something like this? 
```suggestion

    // No "bar" or "*.txt" files. Only copies the following:
```

**Review by mpeddada1 — APPROVED** [review]

**Pull request merged**